src/ILVerify/src/Verifier.cs

```diff
@@ -0,0 +1,255 @@
+        {
+            _typeSystemContext = context;
+        }
+
+        public void SetSystemModuleName(AssemblyName name)
+        {
+            _typeSystemContext.SetSystemModule(_typeSystemContext.GetModule(_typeSystemContext._resolver.Resolve(name)));
+        }
+
+        public IEnumerable<VerificationResult> Verify(PEReader peReader)
+        {
+            if (peReader == null)
+            {
+                throw new ArgumentNullException(nameof(peReader));
+            }
+
+            if (_typeSystemContext.SystemModule is null)
+            {
+                yield return s_noSystemModuleResult;
+                yield break;
+            }
+
+            IEnumerable<VerificationResult> results;
+            try
+            {
+                EcmaModule module = _typeSystemContext.GetModule(peReader);
+                results = VerifyMethods(module, peReader.GetSimpleName(), module.MetadataReader.MethodDefinitions);
+            }
+            catch (VerifierException e)
+            {
+                results = new[] { new VerificationResult() { Message = e.Message } };
+            }
+
+            foreach (var result in results)
+            {
+                yield return result;
+            }
+        }
+
+        public IEnumerable<VerificationResult> Verify(PEReader peReader, TypeDefinitionHandle typeHandle)
+        {
+            if (peReader is null)
+            {
+                throw new ArgumentNullException(nameof(peReader));
+            }
+
+            if (typeHandle.IsNil)
+            {
+                throw new ArgumentNullException(nameof(typeHandle));
+            }
+
+            if (_typeSystemContext.SystemModule is null)
+            {
+                yield return s_noSystemModuleResult;
+                yield break;
+            }
+
+            IEnumerable<VerificationResult> results;
+            try
+            {
+                EcmaModule module = _typeSystemContext.GetModule(peReader);
+                var typeDef = peReader.GetMetadataReader().GetTypeDefinition(typeHandle);
+                results = VerifyMethods(module, peReader.GetSimpleName(), typeDef.GetMethods());
+            }
+            catch (VerifierException e)
+            {
+                results = new[] { new VerificationResult() { Message = e.Message } };
+            }
+
+            foreach (var result in results)
+            {
+                yield return result;
+            }
+        }
+
+        public IEnumerable<VerificationResult> Verify(PEReader peReader, MethodDefinitionHandle methodHandle)
+        {
+            if (peReader is null)
+            {
+                throw new ArgumentNullException(nameof(peReader));
+            }
+
+            if (methodHandle.IsNil)
+            {
+                throw new ArgumentNullException(nameof(methodHandle));
+            }
+
+            if (_typeSystemContext.SystemModule is null)
+            {
+                yield return s_noSystemModuleResult;
```

Comment:
I do not think this should be verification error. It should just throw exception - it is invalid use.